llama-rs/src/vocabulary.rs

```diff
@@ -26,41 +24,23 @@ pub struct Vocabulary {
     pub(crate) max_token_length: usize,
 }
 
-#[derive(Debug, Clone, Error)]
-/// Errors encountered when adding a token to a vocabulary.
-pub enum AddTokenError {
-    #[error("the id of token added should be {expected_id}; is {actual_id}")]
-    /// The token that was added does not have the expected ID.
-    WrongId {
-        /// The expected ID.
-        expected_id: TokenId,
-        /// The actual ID.
-        actual_id: TokenId,
-    },
-}
-
 impl Vocabulary {
     /// Add a token to the vocabulary.
     ///
     /// The token added must have `id` directly after the last token in the vocabulary.
-    pub fn push_token(
-        &mut self,
-        id: TokenId,
-        content: Token,
-        score: TokenScore,
-    ) -> Result<(), AddTokenError> {
+    pub fn push_token(&mut self, id: TokenId, content: Token, score: TokenScore) {
```

Comment:
should probably call this `push_token_unchecked` or something

Reply:
I think panicking here is acceptable - if it panics, there's a bug in the loader and it should be fixed as soon as possible. It's not something that could be meaningfully recovered from as an end-user.

Reply:
Can you change this to `pub(crate)` and change the function name?

Reply:
Sure. I'll keep the function name the same (there's not really a convention around marking functions that can panic in Rust, as far as I know), but I'll mark it `pub(crate)` and add a note to the doc comment about its panic-safety.

Reply:
it's usually called `*_uncheck` or something